lib/API/MP/v1/API.php

```diff
@@ -82,6 +88,23 @@ function getSubscriberFields() {
     return $data;
   }
 
+  function addSubscriberField(array $data = []) {
+    try {
+      $custom_field = CustomField::createOrUpdate($this->custom_fields_data_sanitizer->sanitize($data));
+      $errors = $custom_field->getErrors();
+      if (!empty($errors)) {
+        throw new \Exception('Failed to save a new subscriber field');
+      }
+      $custom_field = CustomField::findOne($custom_field->id);
```

Comment:
Is it really necessary to fetch it again? I think it shouldn't be - when there were no errors in `createOrUpdate` then you should be able to rely on the field being saved and `$custom_field` containing it.

Reply:
It is consistent with other methods in the API. 

Also without this, we wouldn't return all the fields. `created_at` and `updated_at` would not be populated.